pkgs/_macro_client/lib/macro_client.dart

```diff
@@ -36,21 +43,54 @@ class MacroClient {
     return MacroClient._(macros, socket);
   }
 
-  void _send(Map<String, Object?> node) {
+  void _sendRequest(MacroRequest request) {
+    // TODO(davidmorgan): currently this is JSON with one request per line,
+    // switch to binary.
+    socket.writeln(json.encode(request.node));
+  }
+
+  void _sendResponse(Response response) {
     // TODO(davidmorgan): currently this is JSON with one request per line,
     // switch to binary.
-    socket.writeln(json.encode(node));
+    socket.writeln(json.encode(response.node));
   }
 
   void _handleRequest(String request) async {
-    // TODO(davidmorgan): support more than one request type.
-    final augmentRequest =
-        AugmentRequest.fromJson(json.decode(request) as Map<String, Object?>);
-    // TODO(davidmorgan): support multiple macros.
-    final response = await macros.single.augment(host, augmentRequest);
-    _send(response.node);
+    final jsonData = json.decode(request) as Map<String, Object?>;
+    final hostRequest = HostRequest.fromJson(jsonData);
+    switch (hostRequest.type) {
+      case HostRequestType.augmentRequest:
+        _sendResponse(Response.augmentResponse(
+            await macros.single.augment(host, hostRequest.asAugmentRequest)));
+      default:
+      // Ignore unknown request.
```

Comment:
If the deserialization could give you the "unknown" type when it sees something it doesn't recognize, then that could help to ensure you are handling all the known cases at least instead of losing any exhaustiveness checking. But either way 🤷

Reply:
Yes, deserialization gives you the "unknown" type :) ... but I think if we need to use `default` rather then explicitly checking `unknown` in switch statements, so that the switch stays exhaustive when a new type is added. Otherwise it will cause compile errors in existing code to add a new type.

Now that I think about it this is a thing macro authors could run into, we may need to give this some thought ... will file an issue.